src/interviewkit/slicer.py

```diff
@@ -18,102 +18,90 @@
+
+    full_time_value, time_info = zip(*time_info_values)
+
+    time =  datetime.strptime(":".join(full_time_value), ":".join(time_info))
+
+    return timedelta(hours=time.hour, minutes=time.minute, seconds=time.second)
+
+
+def parse_input(argv: list[str]):
+    if len(argv) != 4:
+        raise ValueError("Usage: python3 slicer.py <filepath> <audio start time in minutes> <audio end time in minutes>")
     
-    # Check if audio start and end times are within original audio size limits
-    if(audio_start_time > original_audio_size_ms or audio_end_time > original_audio_size_ms):
-        print("Error! Audio slice input params cannot be greater than original audio size. Please try again with correct parameters.")
-        exit(1)
+    path = Path(argv[1])
+     
+    # Converting audio start and end times in msecs
+    audio_start_time = convert_time_input_to_time_delta(argv[2])
+    audio_end_time = convert_time_input_to_time_delta(argv[3])
+
+    return SlicerInput(path, audio_start_time, audio_end_time)
+
+def create_output_path(input: SlicerInput):
+    start_minutes, start_seconds = divmod(input.start_time.seconds, 60)
+    end_minutes, end_seconds = divmod(input.end_time.seconds, 60)
+    start_time_output = f'{start_minutes}m{start_seconds}s'
+    end_time_output = f'{end_minutes}m{end_seconds}s'
+    new_filepath =   input.path.parent / f"sampled-{start_time_output}-{end_time_output}-{input.path.name}"
+    return new_filepath
 
-    # Audio slicing process
-    audio = audio[audio_start_time:audio_end_time]
+def slice_audio(audio: pydub.AudioSegment, start_time: timedelta , end_time: timedelta):
+
+
+    if(start_time.seconds > audio.duration_seconds or end_time.seconds > audio.duration_seconds):
+        raise ValueError("slice not within audio length")
+
+    start_time_ms = int(start_time.total_seconds() * 1000)
+    end_time_ms = int(end_time.total_seconds() * 1000)
+    
+    return audio[start_time_ms:end_time_ms]
 
-    # Filename for exported file
-    audio_start_time_name = export_filename(audio_start_time_list)
-    audio_end_time_name = export_filename(audio_end_time_list)
-    new_filename =  f"{path.parent}/sampled-{audio_start_time_name}-{audio_end_time_name}-{path.name}"
-    audio.export(new_filename, format="mp3")
-    print("Created new file: ", new_filename)
 
 def main():
-    if len(sys.argv) != 4:
-        print("Usage: python3 slicer.py <filepath> <audio start time in minutes> <audio end time in minutes>")
-        return
 
-    path = Path(sys.argv[1])
-    audio_slice_start_time = (sys.argv[2])
-    audio_slice_end_time = (sys.argv[3])
+    argv = parse_input(sys.argv)
+
+    audio = pydub.AudioSegment.from_file(argv.path)
+
+    try:
+       sliced_audio =  slice_audio(audio, argv.start_time, argv.end_time)
+    except ValueError:
+        raise ValueError("Error! Audio slice input params cannot be greater than original audio size. Please try again with correct parameters.")
```

Comment:
Cool, I like what you did here. Extending the value error message is good. In the future we may want something like an exceptions.py with custom exceptions.